tests/test_task_engine.py

```diff
@@ -1094,6 +1099,25 @@ async def my_task():
             await task_run.state.result()
 
 
+class TestTaskTimeTracking:
+    async def test_start_time_set(self):
+        ID = None
+
+        @task
+        async def foo():
+            nonlocal ID
+            ID = TaskRunContext.get().task_run.id
+
+        await run_task_async(foo)
+        run = await get_task_run(ID)
+
+        assert run.start_time is not None
+        states = await get_task_run_states(ID, StateType.RUNNING)
+        assert len(states) == 1
+        running = states[0]
+        assert running.timestamp == run.start_time
```

Comment:
We should probably do a sync version for completeness?

Reply:
ah that is true